src/main/java/nl/tudelft/ewi/devhub/server/web/resources/AssignmentsResource.java

```diff
@@ -325,9 +325,9 @@ public Response editAssignment(@Context HttpServletRequest request,
         assignment.setName(name);
         assignment.setSummary(summary);
 
-        if(release != null) {
-            assignment.setGradesReleased(release);
-        }
+        //a little bit ugly, form params don't work nicely with checkboxes.
+        boolean releaseStatus = release != null && release.equals("on");
```

Comment:
This fixes the unpublishing issue, I misunderstood the FormParam at first this is ugly but works, they don't do check boxes that nicely.